pkg/scheduler/core/equivalence_cache.go

```diff
@@ -290,3 +294,33 @@ func getEquivalenceHash(pod *v1.Pod) *equivalencePod {
 	// TODO(misterikkit): Also normalize nested maps and slices.
 	return ep
 }
+
+func normalizeContainers(containers []v1.Container) {
```

Comment:
I don't fully understand what is the purpose of this change. Can you please explain?

Reply:
Because `DeepHashObject `considers nil and empty slices to be different. So it should set to be nil if slice is empty. In func `getEquivalenceHash` it only normalize non-nestedmaps and slices. And there is TODO by @misterikkit `Also normalize nested maps and slices.` So I also normalize nested containers slices.